fvm/contractFunctionInvoker.go

```diff
@@ -1,70 +1,79 @@
 package fvm
 
 import (
-	"go.opentelemetry.io/otel/attribute"
-
 	"github.com/onflow/cadence"
 	"github.com/onflow/cadence/runtime"
 	"github.com/onflow/cadence/runtime/common"
 	"github.com/onflow/cadence/runtime/sema"
+	"go.opentelemetry.io/otel/attribute"
 
 	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/module/trace"
 )
 
+// ContractFunctionSpec specify all the information, except the function's
+// address and arguments, needed to invoke the contract function.
+type ContractFunctionSpec struct {
```

Comment:
In the future the address can also be added as a `func(chainID) Address` provider. Since the address should only depend on the chain id

Reply:
yeah, that's reasonable.  i can add that as part of the followup PR